Stellar-contract.x

```diff
@@ -255,6 +258,63 @@ struct Int128Parts {
     uint64 hi;
 };
 
+struct ContractInvocation
+{
+    Hash contractID;
+    SCSymbol functionName;
+};
+
+struct AuthorizedInvocation
+{
+    ContractInvocation callStack<>;
+    SCVec topArgs;
+    uint64* nonce;
+};
+
+enum SCAccountIdType
+{
+    SC_ACCOUNT_ID_TYPE_BUILTIN_INVOKER = 0,
+    SC_ACCOUNT_ID_TYPE_BUILTIN_CLASSIC_ACCOUNT = 1,
+    SC_ACCOUNT_ID_TYPE_BUILTIN_ED25519 = 2,
+    SC_ACCOUNT_ID_TYPE_GENERIC_ACCOUNT = 3
```

Comment:
What's the significance of generic account vs classic account?

Reply:
It sounds like these are:
- classic account = stellar account
- generic account = contract?

Reply:
That's right (to be more precise, generic account = smart wallet contract with a specific interface), and from the general contract perspective everything is just an 'account', hence I wanted to make it clear that none of these entities corresponds to just 'Account'.